dask/dataframe/groupby.py

```diff
@@ -411,6 +702,63 @@ def __getattr__(self, key):
         except KeyError as e:
             raise AttributeError(e)
 
+    @derived_from(pd.core.groupby.DataFrameGroupBy)
+    def aggregate(self, arg, split_every=None):
+        if arg == 'size':
+            return self.size()
+
+        if isinstance(self.index, tuple) or np.isscalar(self.index):
+            group_columns = {self.index}
+
+        elif isinstance(self.index, list):
+            group_columns = {i for i in self.index
+                             if isinstance(i, tuple) or np.isscalar(i)}
```

Comment:
The `group_columns` columns are only used to create the list of `non_group_columns`. Since only `contains` is checked the order of `group_columns` is relevant.